nixos/modules/security/opendoas.nix

```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.security.opendoas;
+
+  inherit (pkgs) opendoas;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    security.opendoas.enable = mkOption {
+      type = types.bool;
+      default = true;
+      description =
+        ''
+          Whether to enable the <command>doas</command> command, which
+          allows non-root users to execute commands as root.
+        '';
+    };
+
+    security.opendoas.configFile = mkOption {
+      type = types.lines;
+      # Note: if syntax errors are detected in this file, the NixOS
+      # configuration will fail to build.
+      description =
+        ''
+          This string contains the contents of the
+          <filename>doas.conf</filename> file.
+        '';
+    };
+
+    security.opendoas.extraConfig = mkOption {
```

Comment:
@joachifm  I am still figuring out the right way to get `/etc/doas.conf`. I saw this in sudo, and was trying it out. I will withdraw this PR until I can figure the right way to do this.